frontend/src/app/Settings/Facility/facilitySchema.ts

```diff
@@ -44,48 +71,60 @@ const providerSchema: yup.SchemaOf<RequiredProviderFields> = yup.object({
 });
 
 export const facilitySchema: yup.SchemaOf<RequiredFacilityFields> = yup.object({
-  name: yup.string().required(),
+  name: yup.string().required("Facility name is missing"),
   cliaNumber: yup
     .string()
-    .required()
-    .test((input, facility) => {
-      if (!stateRequiresCLIANumberValidation(facility.parent.state)) {
-        return true;
-      }
+    .required("CLIA number should be 10 characters (##D#######)")
+    .test(
+      "facility-clia",
+      "CLIA number should be 10 characters (##D#######)",
+      (input, facility) => {
+        if (!stateRequiresCLIANumberValidation(facility.parent.state)) {
+          return true;
+        }
 
-      if (!input) {
-        return false;
-      }
+        if (!input) {
+          return false;
+        }
 
-      return isValidCLIANumber(input);
-    }),
-  street: yup.string().required(),
-  zipCode: yup.string().required(),
-  deviceTypes: yup.array().of(yup.string().required()).min(1).required(),
-  defaultDevice: yup.string().required(),
+        return isValidCLIANumber(input);
+      }
+    ),
+  street: yup.string().required("Facility street is missing"),
+  zipCode: yup.string().required("Facility zip code is missing"),
+  deviceTypes: yup
+    .array()
+    .of(yup.string().required())
+    .min(1, "There must be at least one device")
+    .required("There must be at least one device"),
+  defaultDevice: yup.string().required("A default device must be selected"),
   orderingProvider: providerSchema.nullable(),
   phone: yup
     .string()
-    .test(function (input) {
-      if (!input) {
-        return false;
+    .test(
+      "facility-phone",
+      "Facility phone number is missing or invalid",
+      function (input) {
+        if (!input) {
+          return false;
+        }
+        const number = phoneUtil.parseAndKeepRawInput(input, "US");
```

Comment:
You wrapped this function call in a try/catch in the person schema file, which was smart - want to do that here as well?

Reply:
👍 implemented in fb8d68c